src/SnippetRendererBundle/objects/TPkgSnippetRenderer.class.php

```diff
@@ -147,13 +161,18 @@ public function render()
         }
 
         try {
-            $content = $this->getTwigHandler()->render($this->getSource(), $this->getVars());
-        } catch (Twig_Error $e) {
-            throw new TPkgSnippetRenderer_SnippetRenderingException(
-                sprintf("%s\nin file %s at line %s\n", $e->getMessage(), $e->getFile(), $e->getLine()),
-                $e->getCode(),
-                $e
-            );
+            if ($this->getSourceType() !== IPkgSnippetRenderer::SOURCE_TYPE_STRING) {
+                // use the normal file-only Twig environment
+
+                $content = $this->twigEnvironment->render($this->getSource(), $this->getVars());
+            } else {
+                $content = $this->twigStringEnvironment->render($this->getSource(), $this->getVars());
+            }
+        } catch (Error $e) {
+            $message = sprintf('Error while rendering view %s: %s', $this->getSource(), $e->getMessage());
+            $this->logger->error($message, ['error' => $e]);
+
+            return $message;
```

Comment:
Because the (new) exception would crash the system.

Therefore this and every other exception (they cannot be distinguished) is caught here.

(Besides I see 1000s of usages of this method and none catches an exception).